include/wil/cppwinrt_register_com_server.h

```diff
@@ -0,0 +1,107 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+//! @file
+//! Utilities for making managing COM server easier
+
+#ifndef __WIL_CPPWINRT_REGISTER_COM_SERVER_INCLUDED
+#define __WIL_CPPWINRT_REGISTER_COM_SERVER_INCLUDED
+
+#include <winrt/base.h>
+#include <vector>
+
+namespace wil::details
+{
+template <typename T>
+struct CppWinRTClassFactory : winrt::implements<CppWinRTClassFactory<T>, IClassFactory, winrt::no_module_lock>
+{
+    HRESULT __stdcall CreateInstance(IUnknown* outer, GUID const& iid, void** result) noexcept final
+    {
+        *result = nullptr;
+
+        if (outer)
+        {
+            return CLASS_E_NOAGGREGATION;
+        }
+
+        return winrt::make_self<T>().as(iid, result);
+    }
+
+    HRESULT __stdcall LockServer(BOOL) noexcept final
+    {
+        return S_OK;
+    }
+};
+
+template <typename T = void, typename... Rest>
+void register_com_server(std::vector<DWORD>& registrations)
+{
+    DWORD registration{};
+    winrt::check_hresult(CoRegisterClassObject(
+        winrt::guid_of<T>(), winrt::make<CppWinRTClassFactory<T>>().get(), CLSCTX_LOCAL_SERVER, REGCLS_MULTIPLEUSE, &registration));
+    registrations.push_back(registration);

```

Comment:
Maybe getting too nit-picky, but if this throws, you'll leak the registration. Do we have a pre-existing RAII type for this in WIL? Something like `unique_coclass_cookie`?

Reply:
Actually, with that, you wouldn't even need the `com_server_revoker` type. Just return a `std::vector<wil::unique_coclass_cookie>` and you get effectively the same thing. Revoking is just a call to `.clear()`

Reply:
I think push back is no-throw if a previous call to `reserve` succeeds and the number of element to push < capacity?

Reply:
Gotcha. I missed the `reserve` call because it was in a different location. Worth a comment here.

Reply:
@dunhor Added comment. I find I like the idea of `unique_coclass_cookie` though, should I add it in another PR or? I'm just afraid `unique_coclass_cookie` alone would take another round of like 20+ comments😅

Reply:
> should I add it in another PR or?

Looking now since I didn't initially, looks like `unique_com_class_object_cookie` already exists and is probably what you want here

Reply:
Updated to return `std::vector<wil::unique_com_class_object_cookie>` instead of `com_revoker`.